src/tokenizer.ts

```diff
@@ -168,182 +177,352 @@ export enum Token {
 
   // meta
 
+  DIGIT,
+  MAYBE_KEYWORD,
+  WHITESPACE,
+  COMMENT_OR_OPERATOR,
+  OPERATOR,
   INVALID,
-  ENDOFFILE
+  EOF
 }
 
-export enum IdentifierHandling {
+export const enum IdentifierHandling {
   DEFAULT,
   PREFER,
   ALWAYS
 }
 
-export function tokenFromKeyword(text: string): Token {
+// Classify single character tokens (0..127)
+const BASIC_TOKENS: Token[] = [
```

Comment:
I think this table could become hard to maintain as it is not immediately obvious how to approach it for someone with little experience with the code base, and can easily be missed during PRs by those who would know. Any ideas?

Reply:
I don't think this file will be changed very often. Unless there is a new keyword and you have to change IDENTIFIER to MAYBE_KEYWORD / IDENTIFIER_OR_KEYWORD somewhere. Otherwise, the table is already pretty much full and will not be changed in the future, as this would break backwards compatibility of syntax. So I don't see a problem